courseware/api.py

```diff
@@ -198,6 +198,42 @@ def create_edx_auth_token(user):
     return auth
 
 
+def update_edx_user_email(user):
```

Comment:
It's test is already covered here https://github.com/mitodl/mitxpro/pull/1543/files#diff-a38b752386fe52b310bddffcb0688689161

As `change_edx_user_email_async ` is calling the API, so I have mocked `requests Session`. If you see the coverage its already covered here.

Reply:
It's import to distinguish that coverage doesn't mean that the behavior of the code is actually tested, but it really just means the code was called somewhere in a test.

In this case, because you're mocking the entire requests session, the only thing you're testing is that `update_edx_user_email` doesn't raise an exception if `requests.Session` is a mock instance. As an example, you can see that [this test](https://github.com/mitodl/mitxpro/blob/master/courseware/api_test.py#L142) is asserting that the function `create_edx_auth_token` is making the correct calls. Normally we wouldn't worry so much about the internal behavior of a function in such details and we'd only concern ourselves with the return value and/or database state, but since this is an integration boundary over to openedx, it's important to assert that behavior so we know when it breaks.

Regarding the coverage in `views_test.py, as a best practice it's important to separate concerns -  tests should also only be testing the behavior of the module the tests are for so that those concerns are separated, so `views_test.py` shouldn't be testing code in `api.py`. Since the call has been moved into the task, the test is actually leaking across 2 module boundaries: `views.py` -> `tasks.py` -> `api.py`. It can sometimes feel like you're testing the same thing twice, but you're actually asserting 2 different levels of logic.

I'll leave more detailed comments on the rest of the PR explaining how best to deal with that so we can keep those concerns separated but still test the code appropriately.